### ISSUE TITLE
Setting keep_alive to 0 causes a BlockingIOError Exception due to socket Timeout

### ISSUE THREAD

**Issue by schneeemensch** [eclipse/paho.mqtt.python#506]
In version 1.5.0 it was added that the socket default timeout is set by the keep_alive value.

As discussed in #426 this can have implications on users who have set keep_alive to 0. 
According to the MQTT and paho.mqtt documentation this should disable the keep_alive mechanism.
We are using this in send-only clients where we cannot spawn an additional thread and cannot guarantee that they are accessed frequently.

Starting with version 1.5.0 this causes the socket connect to always timeout.
Either an additional option should be found to define the socket timeout or the default timeout should be used for cases where keep_alive=0. The first option is better for multiple reasons.

Here is the error output:

```
     self.mqtt_client.connect(self.mqtt_server, self.mqtt_port, self.keep_alive)
   File "/localdev/clemens/tif/workspaces/ws1/audi_tif/venv/lib/python3.7/site-packages/paho/mqtt/client.py", line 937, in connect
     return self.reconnect()
   File "/localdev/clemens/tif/workspaces/ws1/audi_tif/venv/lib/python3.7/site-packages/paho/mqtt/client.py", line 1071, in reconnect
        sock = self._create_socket_connection()
   File "/localdev/clemens/tif/workspaces/ws1/audi_tif/venv/lib/python3.7/site-packages/paho/mqtt/client.py", line 3522, in _create_socket_connection
     return socket.create_connection(addr, source_address=source, timeout=self._keepalive)
   File "/global/freeware/Linux/RHEL7/python-3.7.0/lib/python3.7/socket.py", line 727, in create_connection
     raise err
   File "/global/freeware/Linux/RHEL7/python-3.7.0/lib/python3.7/socket.py", line 716, in create_connection
     sock.connect(sa)
BlockingIOError: [Errno 115] Operation now in progress
```

**Comment by schneeemensch** [comment]
We have now locally patched paho.mqtt to use the default socket timeout when keep_alive is set to 0:

Here are the diffs:

![image](https://user-images.githubusercontent.com/44439378/91708925-4e4a7600-eb82-11ea-8a3c-8de4bc669787.png)

![image](https://user-images.githubusercontent.com/44439378/91708810-22c78b80-eb82-11ea-9c65-90cc13793864.png)

**Comment by babaMar** [comment]
Any chance this can be made in a PR?

**Comment by schneeemensch** [comment]
I have checked if this is fixed with the latest release (1.6.1) and our issue is gone.
As far as I see the timeout for the `socket.create_connection` call is now using a fixed timeout of 5 seconds, which seems to be enough in our use case.

I would therefore close this issue.
